src/site/_collections/algolia.js

```diff
@@ -0,0 +1,185 @@
+const striptags = require('striptags');
+const stripcomments = require('strip-comments');
+const removeMarkdown = require('remove-markdown');
+
+const AuthorsCollection = require('./authors');
+const TagsCollection = require('./tags');
+const {defaultLocale} = require('../_data/site');
+const {livePosts} = require('../_filters/live-posts');
+const {getDefaultUrl} = require('../_filters/urls');
+const {generateSrc} = require('../_includes/components/Img');
+const {supportedLocales} = require('../../../shared/locale');
+
+/**
+ * Cleans text and shrink the size of the given fulltext to fit within a certain limit,
+ * at the nearest found newline character.
+ *
+ * @param {string} content
+ * @param {number} [limit]
+ * @return {string}
+ */
+function limitText(content, limit = 7500) {
+  // Remove all comments
+  content = stripcomments(content);
+  // Remove all html
+  content = striptags(content);
+  // Remove all markdown
+  content = removeMarkdown(content);
+  // Remove extra spaces
+  content = content.replace(/\s+/g, ' ');
+
+  if (content.length <= limit) {
+    return content;
+  }
+
+  // Find the nearest prior newline to the 10k limit.
+  let newlineIndex = content.lastIndexOf('\n', limit);
+  if (newlineIndex === -1) {
+    newlineIndex = limit;
+  }
+  return content.slice(0, newlineIndex);
+}
+
+/**
+ * @param {EleventyCollectionObject} collections
+ * @returns {AlgoliaCollection}
+ */
+module.exports = (collections) => {
+  /** @type {AlgoliaCollection} */
+  const algoliaCollection = [];
+  /** @type {{
+    [url: string]: {
+      [locale: string]: AlgoliaCollectionItem;
+    };
+  }} */
+  const algoliaCollectionProcessing = {};
+  // if (process.env.ELEVENTY_ENV !== 'prod') {
+  //   return algoliaCollection;
+  // }
+
+  /**
+   * @param {AlgoliaCollectionItem} algoliaCollectionItem
+   */
+  const addToCollection = (algoliaCollectionItem) => {
+    if (!algoliaCollectionProcessing[algoliaCollectionItem.url]) {
+      algoliaCollectionProcessing[algoliaCollectionItem.url] = {};
+    }
+
+    algoliaCollectionProcessing[algoliaCollectionItem.url][
+      algoliaCollectionItem.locales[0]
+    ] = algoliaCollectionItem;
+  };
+
+  const allSorted = collections.getAllSorted().filter((item) => {
+    return item.data.title && item.data.page.url && livePosts(item);
+  });
+  /** @type Array<AuthorsItem|TagsItem> */
+  const virtualCollections = [
+    ...Object.values(AuthorsCollection(collections)),
+    ...Object.values(TagsCollection(collections)),
+  ];
+
+  for (const item of allSorted) {
+    if (item.data.disable_algolia || item.data.noindex) {
+      continue;
+    }
+
+    const defaultUrl = getDefaultUrl(item.url);
+
+    /** @type {AlgoliaCollectionItem} */
+    const algoliaCollectionItem = {
+      title: item.data.title,
+      description: item.data.description,
+      content: limitText(item.template.frontMatter.content),
+      url: defaultUrl,
+      tags: item.data.tags || [],
+      locales: [item.data.lang],
```

Comment:
would anything ever have more than one locale? should this be `locale` instead?

Reply:
Additional locales get added near line 185, a post that there isn't a polish version of will have `pl` add to the locales of the english (default locale) version.